### ISSUE TITLE
Add a GUI application for Julia version management

### ISSUE THREAD

**Issue by davidanthoff** [JuliaLang/juliaup#54]
We could also add a GUI application named `Juliaup` or `Julia Configuration` or something like that. This would appear in the Start menu and would show a list of all channels on the system, which ones are out-of-date, with a button to update individual or all channels.

**Comment by StefanKarpinski** [comment]
If this is going to be done, shouldn't it be separate from `juliaup`? The command line tool seems like it should be as simple and reliable as possible, which is very much at odds with having a GUI.

**Comment by uncomfyhalomacro** [comment]
As I am exploring some issues here and there, I noticed this. I think this is a good idea for new users (or some users) that do not like to setup/manage/update a project or a julia version through the CLI. It's worth noting there are a lot of possible GUI frameworks we can use in Rust.

- gtk-rs
- iced-rs
- tauri
- slint

All of these are platform agnostic. Slint and tauri seems promising though.

**Comment by davidanthoff** [comment]
> If this is going to be done, shouldn't it be separate from juliaup?

Yes, at least initially I definitely think so. If something emerges that is super reliable and feature complete etc I could imagine that we bundle it down the road with the stuff here, but I'm more thinking years than months ;)

I think architecturally what we need is to expand the API that we have in Juliaup already to cover all the cases that a UI needs, and then the UI can use that under the hood to implement the actual functionality. VS Code could then also expand its integration with Juliaup via the same API (it already uses the current, pretty meager API to find Julia versions).
